pandas/core/series.py

```diff
@@ -159,6 +163,8 @@
     from pandas.core.groupby.generic import SeriesGroupBy
     from pandas.core.resample import Resampler
 
+    SeriesT = TypeVar("SeriesT", bound="Series")
```

Comment:
> As we look to make the types public, a follow-up to use SeriesT (and DataFrameT) throughout maybe worthwhile.

@twoertwein sorry to mess you around. we have an issue which we have not yet addressed.

from https://github.com/pandas-dev/pandas/issues/32638#issuecomment-598044823

> For example, in GeoPandas, an operation like reindex can remove the geometry column, in which case it makes sense to return a normal DataFrame.

so for now let's not put any restrictions on the return type being the same as the class. And just use Series/DataFrame for now. i.e. remove the typevars added in this PR.

Reply:
I'm happy to do that. Personally, I would slightly prefer the TyperVar (more consistent with NDFrame). GeoPandas would probably anyway need an ignore comment (Union of their sub-class or Series).

Reply:
I agree with DataFrame and Series since that is what the implementation does:

```py
import pandas as pd

class MyDataFrame(pd.DataFrame):
    ...

class MySeries(pd.Series):
    ...

print(type(MyDataFrame().sort_index()))  # DataFrame
print(type(MySeries().sort_index())) # Series
```